strawberryfields/engine.py

```diff
@@ -291,15 +298,20 @@ def _run_program(self, prog, **kwargs):
             except NotImplementedError:
                 # command not directly supported by backend API
                 raise NotImplementedError('The operation {} has not been implemented for {}.'.format(cmd.op, self.backend)) from None
+                # TODO: may need more care in this message. There are occasions where an operation is supported,
```

Comment:
fyi @smite @josh146

Reply:
It should be relatively simple to allow for the Dgate+MeasureFock to be used in combination, by modifying the sampling routine in the Hafnian library to use loop hafnians (cc @nquesada, just in case I am over simplifying). In this case, it should solve this problem, assuming there are no other combinations that don't work together